### PR TITLE
fix: show response headers w/o mediaType content

### PR DESCRIPTION
**Pull request by julshotal**
## 🧰 Changes

If we're creating response examples and there's no examples/we can't generate examples because there's no content but there's headers
- Status codes without content that have headers now get an empty `mediaType ` with the type of wildcard `*/*`
- A boolean will get passed on these statuses indicating that they only have headers (which will be used on the front end)

The handling of populating an example for these using the status code + status message [will take place in our Response react component](https://github.com/readmeio/readme/pull/6101) 👍 This PR is to just get the data there 

## 🧬 QA & Testing
Tests are passing

### CODE REVIEW

**Comment by Dashron** [comment]
let's say someone legit doesn't return a response (e.g. 204 status code), with headers. How is that going to show up in this case? will it indicate that it returns json with an empty object?

That could be an issue.

File: __tests__/__datasets__/operation-examples.json [hunk 1]
```diff
@@ -684,6 +678,36 @@
           }
         }
       }
+    },
+    "/headers-but-no-content": {
+      "post": {
+        "description": "This operation has request and response headers but no content or media types",
+        "requestBody": {
+          "headers": {
```
[the remark this answers is not in the document]
Comment (erunion) [reply __tests__/__datasets__/operation-examples.json:686]: (thought i sent that last night, apparently didnt)

[the remark this answers is not in the document]
Comment (julshotal) [reply __tests__/__datasets__/operation-examples.json:686]: I checked, it is required. Going to rework this example while looking into Aaron's question

**Comment by julshotal** [comment]
> let's say someone legit doesn't return a response (e.g. 204 status code), with headers. How is that going to show up in this case? will it indicate that it returns json with an empty object?
> 
> That could be an issue.

Fixing this issue by using a wildcard mediaType & handling the example on the front end to use a status code + message

File: __tests__/__datasets__/operation-examples.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (erunion) [reply __tests__/__datasets__/operation-examples.json:695]: Looking at this again and `headers` isn't valid here for request bodies.

```
> npx rdme validate test.yaml 

OpenAPI schema validation failed.

ADDITIONAL PROPERTY must NOT have additional properties

  24 |             }
  25 |           },
> 26 |           "headers": {
     |           ^^^^^^^^^ 😲  headers is not expected to be here!
  27 |             "request-header": {
  28 |               "description": "A request header",
  29 |               "style": "simple",
```

https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.3.md#request-body-object

**Review by erunion — APPROVED** [review]

**Comment by julshotal** [comment]
Don't know what's up with that date test failure, looking into it

**Comment by erunion** [comment]
@julshotal I fixed that earlier this week in https://github.com/readmeio/oas/pull/585/files#r791171564, if you rebase this branch it should work again

File: __tests__/__datasets__/operation-examples.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (julshotal) [reply __tests__/__datasets__/operation-examples.json]: Okay I _think_ its okay now 🤡

[the remark this answers is not in the document]
Comment (erunion) [reply __tests__/__datasets__/operation-examples.json]: <sub><sub><sub>sorry</sub></sub></sub>

**Pull request merged**